### DISCUSSION TITLE
ValueError n_components=128 must be between 0 and min(n_samples, n_features)=2 with svd_solver='full'

### DISCUSSION THREAD

**Discussion by lang216** [acids-ircam/RAVE#125] [Q&A] [answered]
Hi, 

I was using the google Colab provided and tried to train on some sounds. However, after running the training block, I always get the error ValueError n_components=128 must be between 0 and min(n_samples, n_features)=32 with svd_solver='full'. 

I think it has something to do with the PCA algorithm, more specifically, the number of samples I used. Do I have to use more 128 samples in order to make this model work?

Best,
Lang Chen

**Comment by ghost** [comment, marked as answer]
i had similiar issue and i think it was because my training dataset was too small / not enough audio material https://github.com/acids-ircam/RAVE/issues/108

**Reply by lang216** [reply]
Yes, this is the issue! We need to use much more samples (larger training set)!

**Reply by clusternd** [reply]
Have you figured out approximate minimum size of training dataset? I've just run into the same problem

**Reply by lang216** [reply]
I think you need more than 128 samples. Also, you might want your total audio duration to be more than 3 hours in order to have a good result.
